Library/Homebrew/dev-cmd/bump-formula-pr.rb

```diff
@@ -0,0 +1,73 @@
+require "formula"
+
+module Homebrew
+  def bump_formula_pr
+    formula_arg = ARGV.named.first
+    odie "No formula specified!" unless formula_arg
+    formula = Formula[ARGV.named.first]
+    odie "No #{formula} formula found!" unless formula
+
+    requested_spec, formula_spec = if ARGV.include?("--devel")
+      devel_message = " (devel)"
+      [:devel, formula.devel]
+    else
+      [:stable, formula.stable]
+    end
+    odie "#{formula}: no #{requested_spec} specification found!" unless formula
+
+    hash_type, old_hash = if (checksum = formula_spec.checksum)
+      [checksum.hash_type.to_s, checksum.hexdigest]
+    end
+
+    new_url = ARGV.value("url")
+    new_hash = ARGV.value(hash_type)
+    new_tag = ARGV.value("tag")
+    new_revision = ARGV.value("revision")
+    new_url_hash = if new_url && new_hash
+      true
+    elsif new_tag && new_revision
+      false
+    elsif !hash_type
+      odie "#{formula}: no tag/revision specified!"
+    else
+      odie "#{formula}: no url/#{hash_type} specified!"
+    end
+
+    Utils::Inreplace.inreplace(formula.path) do |s|
+      if new_url_hash
+        old_url = formula_spec.url
+        s.gsub!(old_url, new_url)
+        s.gsub!(old_hash, new_hash)
+      else
+        resource_specs = formula_spec.specs
+        old_tag = resource_specs[:tag]
+        old_revision = resource_specs[:revision]
+        s.gsub!(old_tag, new_tag)
```

Comment:
Since the tag is usually version string. This replacement may happen in the wrong places, e.g resources.